### ISSUE TITLE
Metrics and state endpoints

### ISSUE THREAD

**Issue by achamayou** [microsoft/CCF#1183]
Expose a metrics endpoint (based on https://microsoft.github.io/CCF/master/users/rpc_api.html?highlight=status#getmetrics) with throughput and latency information, for service monitoring purposes. We also probably want to expose things like that conflict rates/counts.

**Comment by achamayou** [comment]
After discussion, we probably want multiple different metric endpoints to cover different aspects of the service.

Consensus:

1. `/commit`, which exposes committed `seqno` and `view` on the node
2. `/status`, which exposes the status of the node (also see #1052)
2. `/topology`, which exposes the node's knowledge of the service topology (ie. primary in current config, active configs if primary etc)

Resource usage:

1. `/memory` which should expose the total vs used amount of memory

Endpoint-level metrics:

For each business endpoint, we want to expose:

1. number of requests
2. number of successful responses (2xx. 3xx)
3. number of error responses (4xx)
4. number of failure responses (5xx)

We will probably want to add more detail to this over time (request/response size and latency distribution over a rolling period), but this should be enough to start with.

**Comment by lynshi** [comment]
Will a user or member identity be required to communicate with these endpoints, or can the client be unauthenticated? In particular, `/status` will have to be open because a joining node has no access to the identity tables if it's not part of the network yet. In general, it's more convenient to not have to provide certificates for these as well as long as they're read-only.

**Comment by achamayou** [comment]
@lynshi I don’t think we want to require authorisation, they don’t give access to any confidential information. We can change that easily if need be.

**Comment by achamayou** [comment]
Update: `/memory` is blocked on the necessary API being available in Open Enclave.

`/topology` or perhaps `consensus/config` is queued behind current work on raft.

**Comment by achamayou** [comment]
`/config` is there now, but `/memory` is pending the next Open Enclave release (changes are in master now).
